### DISCUSSION TITLE
Enabling structural knowledge management using choral explanations

### DISCUSSION THREAD

**Discussion by sander-cb** [alkem-io/alkemio#913] [Ideas] [answered]
Thank you for building Alkemio. I am excited to try it out for some of my collaborations on building a trustworthy online environment. In one of these collaborations, I chair a knowledge management working group in a public-private collaboration.

In the current state, we want to start out with the [Choral Explanations](https://hapgood.us/2016/05/13/choral-explanations/) modality, similar to the Stack Overflow Q&A format. This makes for a lower barrier to entry for participants than for example a greenfield wiki or document management system would have. Also the modality encourages community members to engage personally, in their own voice, and potentially with multiple modalities.

My current experiment is in my Demo Hub under the callout “Kennisvragen en -antwoorden” (“Knowledge questions and answers’). Each knowledge question is a card, and each answer is a comment.

To make the platform more fit for use in this context, here I collect some changes I’d currently like to see, in order of preference:

1. Raising the comment limit from its current 500 characters.
2. Enable full data export for a callout or for a full Hub using an API. A potential low-effort implementation could be to enable users to create a personal access token (like on GitHub) to call the GraphQL API directly. Alternatively, since you are on the Ory stack for IAM, I can share my experiences using Ory Hydra for OAuth 2.0. Tracking issue: https://github.com/alkem-io/alkemio/issues/247
3. Ability to edit comments while keeping version history. Ideally I can authorize a group of community members to edit other people’s comments as well.
4. Render the full title of a card instead of just the part that fits on a single line.
5. Support uploading images to cards and comments. Tracking issue: https://github.com/alkem-io/alkemio/issues/667
6. Support embedding images and videos to encourage sharing knowledge using various media.
7. Ability to select an explicit Creative Commons (in our case, CC BY 4.0 or CC0 1.0) license for a callout, requiring contributors to provide their contributions under this license. Make this explicit in the content submission form. This enables reuse of the knowledge in this knowledge base.
8. Ability to upvote cards and comments, such as on Stack Overflow.
9. Ability to respond to a comment. I see no need for more levels of threading, as this can easily become confusing. Tracking issue: https://github.com/alkem-io/alkemio/issues/871
10. Remove cover photo placeholders on cards, in order to not “punish” contributions which have no visual material.
11. Enable reordering callouts on the Explore tab. Tracking issue: https://github.com/alkem-io/alkemio/issues/758
12. Enable callout descriptions to consist of multiple paragraphs. Currently the editor supports this, but all is rendered as a single paragraph.
13. Provide Unsplash search or other easily selectable cover photos for cards.

I’m looking forward to discussing this use case in person soon. If that works for you, we can continue using this GitHub thread to centrally track implementation of this use case, linking to separate tickets for the specific discussions and work.

**Comment by techsmyth** [comment, marked as answer]
@sander-cb thanks for the super detailed set of suggestions..

And nice to have been able to talk through the points of feedback face to face. As covered, quite some of the UI related issues should be addressed in the epic currently in progress: https://github.com/alkem-io/alkemio/issues/838

Addressing your points:
1. This becomes less relevant for your use case one you switch to using callouts to raise the questions. New issue is raised to allow contributors to create callouts: https://github.com/alkem-io/server/issues/2372
2. Data is exportable if you have the access rights using graphql api
3. Edit comments is a design question, we use Matrix Protocol for communications and in there comments never get deleted, though there can be a display of content that makes it appear that way on the client. For further discussion.
4. Covered in UX redesign, suggest you review once that is live
5. On the backlog
6. On the backlog
7. Agreements in general for communities is a big topic, and we are just kicking off a project on this one. Happy to involve you once we get further with it. 
8. This would be great, something that will come up rapidly on the backlog priority. 
9. Raised as part of the issue you linked
10. Part of refreshed UX design
11. Part of refreshed UX design
12. This sounds like a bug. Can you please raise one?
13. Part of the VIsuals refresh epic: https://github.com/alkem-io/alkemio/issues/601

If you are ok with the feedback above can you please mark this thread as solved? In general probably also good to take some of the detailed points into separate items.

Cheers, Neil

**Reply by sander-cb** [reply]
Thanks @techsmyth! I’ll mark the thread as solved and use this post for my own reference to track issues related to this use case. Just for reference:

- Re: 3. Given alkem-io/server#2372 is planned, and as soon as members can edit callouts, I have no more use case for comment editing.
- Re: 12. Will do.

**Reply by sander-cb** [reply]
Issued the bug in 12: https://github.com/alkem-io/alkemio/issues/927

**Reply by sander** [reply]
Following up on this thread in https://github.com/alkem-io/alkemio/issues/936
